### PR TITLE
feat(title): add support for small title

### LINKED ISSUE #18898: feat: iOS toolbar small title
# Feature Request

**Ionic version:**

[x] **4.x**

**Describe the Feature Request**

Add the iOS 13 subtitle to the toolbar:

"Choose a contact to add to Favorites"

Screen Shot 2019-07-25 at 5 28 04 PM

**Related Code**

**Additional Context**

### PR DESCRIPTION
**Pull request by brandyscarney**
✅ Ready to Merge ✅

## Pull request checklist

Please check if your PR fulfills the following requirements:
- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been reviewed and added / updated if needed (for bug fixes / features)
- [x] Build (`npm run build`) was run locally and any changes were pushed
- [x] Lint (`npm run lint`) has passed locally and any fixes were made for failures

## Pull request type

 

 

Please check the type of change your PR introduces:
- [ ] Bugfix
- [x] Feature
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Documentation content changes
- [ ] Other (please describe): 

## What is the current behavior?

Issue Number: closes #18898

## What is the new behavior?

- Adds support for `small` size in title
- Updates styles on the toolbar / searchbar for iOS 13

Screen Shot 2019-08-28 at 4 01 10 PM

| `master` | `branch` |
| ---| ---|
| ios master sizes | ios branch sizes |
| md master sizes | md branch sizes |

## Does this introduce a breaking change?

- [ ] Yes
- [x] No

## Other information

Usage information:

```
<ion-header>
  <ion-toolbar>
    <ion-title size="small">Choose a contact to add to Favorites</ion-title>
  </ion-toolbar>
  <ion-toolbar>
    <ion-title>Contacts</ion-title>
    <ion-buttons slot="end">
      <ion-button>Cancel</ion-button>
    </ion-buttons>
  </ion-toolbar>
  <ion-toolbar>
    <ion-searchbar></ion-searchbar>
  </ion-toolbar>
</ion-header>
```

### CODE REVIEW

File: core/src/components/searchbar/searchbar.tsx [hunk 1]
```diff
@@ -176,6 +182,10 @@ export class Searchbar implements ComponentInterface {
     });
   }
 
+  componentWillLoad() {
```
Comment (manucorporat) [inline core/src/components/searchbar/searchbar.tsx]:
```suggestion
  connectedCallback() {
```

so if moved to different places, we can still re-fire

File: core/src/components/segment/segment.tsx [hunk 2]
```diff
@@ -54,6 +54,7 @@ export class Segment implements ComponentInterface {
 
   /**
    * Emitted when the styles change.
+   * @internal
```
Comment (manucorporat) [inline core/src/components/segment/segment.tsx]: ❤️

File: core/src/components/title/test/sizes/e2e.ts [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+  const compare = await page.compareScreenshot();
+  expect(compare).toMatchScreenshot();
+});
+
+test('title:rtl: sizes', async () => {
+  const page = await newE2EPage({
+    url: '/src/components/title/test/sizes?ionic:_testing=true&rtl=true'
+  });
+
+  await page.waitFor(250);
```
Comment (manucorporat) [inline core/src/components/title/test/sizes/e2e.ts]: not needed anymore

File: core/src/components/title/test/sizes/e2e.ts [hunk 4]
```diff
@@ -0,0 +1,23 @@
+import { newE2EPage } from '@stencil/core/testing';
+
+test('title: sizes', async () => {
+  const page = await newE2EPage({
+    url: '/src/components/title/test/sizes?ionic:_testing=true'
+  });
+
+  await page.waitFor(250);
```
Comment (manucorporat) [inline core/src/components/title/test/sizes/e2e.ts]: not needed

File: core/src/components/title/title.tsx [hunk 5]
```diff
@@ -20,6 +20,29 @@ export class ToolbarTitle implements ComponentInterface {
    */
   @Prop() color?: Color;
 
+  /**
+   * The size of the toolbar title.
+   * Only applies in `ios` mode.
+   */
+  @Prop() size: 'large' | 'default' | 'small' = 'default';
```
Comment (manucorporat) [inline core/src/components/title/title.tsx]: replace 'default'  with undefined?

File: core/src/components/segment/segment.tsx (hunk 2 above)
Comment (brandyscarney) [reply core/src/components/segment/segment.tsx]: :wink:

File: core/src/components/title/title.tsx [hunk 6]
```diff
@@ -22,9 +22,8 @@ export class ToolbarTitle implements ComponentInterface {
 
   /**
    * The size of the toolbar title.
-   * Only applies in `ios` mode.
    */
-  @Prop() size?: 'large' | undefined;
+  @Prop() size?: 'large' | 'small';
```
Comment (liamdebeasi) [inline core/src/components/title/title.tsx]: should this be

```suggestion
  @Prop() size?: 'large' | 'small' | undefined;
```

instead?

**Pull request merged**